18_regex/slides.md

```diff
@@ -936,14 +941,13 @@ If you are using the same regex many times consider compiling it once and cachin
 
 ```scala
 // Recompiles the regex on every iteration
-strings.filter(_.matches("boban.+jones"))
+strings.findFirstIn(_.matches("boban.+jones"))
```

Comment:
I think you misunderstood me when we were talking through this.

This code won't compile. `findFirstIn` is a regex function.

```scala
@ val regex = "\\d+".r 

@ regex.findFirstIn("abc0123") 
// Some("0123")
```

I was saying that it would be nice to have an example that transfers nicely from an uncompiled to compiled form with minimal code changes. That way it's clearer to the audience that the main change is compiling it.

```scala
// Uncompiled example
strings.filter(_.someRegexFunction("boban.+jones"))

// Compiled example
val compiled = "boban.+jones"
strings.filter(_.someRegexFunction(compiled))
```

In my example above, it's easier to get them to focus on the change being about compilation.

Unfortunately when you go from uncompiled to compiled, the api's look completely different. For example `String.matches` only takes uncompiled regexes (don't think there's a compiled equivalent). And `findFirstIn` only takes compiled regexes. So transforming it into a valid code snippet also changes a lot of the surrounding code.

That's why I told you to just change it to `...` to make it less distracting.

My snippet above might be fine actually - it gets the point across and it's clear that it's a made up function and they shouldn't expect it to compile.

Reply:
okayy :) will change the example to the new example

```
---

# Compilation

Compilation takes time

```scala
// Uncompiled example
strings.filter(_.someRegexFunction("boban.+jones"))

// Compiled example
val compiled = "boban.+jones".r
strings.filter(_.someRegexFunction(compiled))
```

---
```

Added a missing `.r` at the back of the compiled boban jones string

Reply:
> Added a missing `.r` at the back of the compiled boban jones string

Haha yeah, I'm getting forgetful in my old age